msprime/ancestry.py

```diff
@@ -1146,6 +1146,128 @@ def sim_ancestry(
+    :param float time: The desired period of time to continue simulation in msprime,
+        in units of generations
+    :param int sample_size: Sample size for :meth:`msprime.simulate`,
+        in units of haploid samples,
+    :param dict kwargs: Any other arguments to :meth:`msprime.simulate`.
+    :type nodes: :class`numpy.ndarray`
+    :param continue_nodes: node IDs in ts to match up to the lineages in the continued
+        simulation, defaults to the samples
+    :return: The :class:`tskit.TreeSequence` object resulting from the
+        union of the input tree equence and the continued simulatoin
+    :rtype: :class:`tskit.TreeSequence`
+    """
+    if continue_nodes is None:
+        continue_nodes = ts.samples()
+    if sample_size is None:
+        sample_size = len(continue_nodes)
+
+    new_ts = simulate(sample_size=sample_size, end_time=time, **kwargs)
+
+    new_nodes = np.where(new_ts.tables.nodes.time == time)[0]
+
+    if len(new_nodes) > len(continue_nodes):
+        raise RuntimeError(
+            f"""More lineages present in new simulation samples than in the old.. \
+            Decrease sample_size to try again. \
+            Setting sample_size={len(continue_nodes)} \
+            or leaving it to default will ensure compliance."""
+        )
+
+    if len(new_nodes) == 0:
+        oldest_node_time = max(
+            [
+                new_ts.node(new_ts.first().roots[i]).time
+                for i in range(0, len(new_ts.first().roots), 1)
+            ]
+        )
+        raise RuntimeError(
+            f"""No surviving lineages in new simulation at merge time. \
+            Depth of new simulation is {oldest_node_time} - alter parameters \
+            to extend simulation or reduce the simulation time."""
+        )
+
+    node_map = np.repeat(tskit.NULL, new_ts.num_nodes)
+    node_map[new_nodes] = np.random.choice(
+        continue_nodes, len(new_nodes), replace=False
+    )
+
+    tables = _adjust_tables_time(ts, time)
+    tables.union(
+        new_ts.tables, node_map, add_populations=False, check_shared_equality=False
+    )
+
+    return tables.tree_sequence()
+
+
+def _adjust_tables_time(ts, time):
+    """
+    Returns a new :class`tskit.tables.TableCollection` by copying all values
+    from a tree sequence's tables and modifying the time in the nodes and
+    mutations tables.
+
+    This method executes code from the pyslim tutorial  'Following up with more
+    coalescent simulation'
+
+    :type ts: :class`tskit.TreeSequence`
+    :param ts: tree sequence used as basis for new tables
+    :param time int: The time to adjust tables by,
+        in units of generations
+    :return: The :class:`tskit.TableCollection` object with modified time
+    :rtype: :class:`tskit.TableCollection`
+    """
+    tables = ts.tables
+    tables.nodes.set_columns(
+        flags=ts.tables.nodes.flags & ~np.uint32(tskit.NODE_IS_SAMPLE),
```

Comment:
oh, right - this code is doing more than shifting the times, anyhow - it's also removing sample flags. We should actually remove not *all* the sample flags, but only the flags from `continue_nodes`.

Reply:
That makes more sense - I wrote this up in a way that makes sense to me but if this is something that is done elsewhere in a specific way I'd be happy to plug that in!